### PR TITLE
Risk app

### PR DESCRIPTION
**Pull request by rat6666**

### CODE REVIEW

File: risk/src/app/shared/interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export interface Risk {
+  id: string;
+  userID: string;
+  riskName: string;
+  time: number;
+  probability: number;
+  description: string;
+}
+
+export interface Calc {
```
Comment (ClainAtomi) [inline risk/src/app/shared/interfaces.ts:16]: do not use abbreviatures in class names,
add more meaningfull name

File: risk/src/app/shared/layouts/risk-edit/risk-edit.component.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  public updateEditedRisk(): void {
+    this.dataBaseService.updateRiskList(this.risk);
+  }
+
+  public resetEditedRisk(): void {
+    this.selectedRiskService.selectRisk(this.initialRiskData);
+  }
+
+  public calcRisk(type: string): void {
+    switch (type) {
```
Comment (ClainAtomi) [inline risk/src/app/shared/layouts/risk-edit/risk-edit.component.ts:66]: split this method into set of small meaningfull ones

File: risk/src/app/shared/layouts/risks-list/risks-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  public selectedRisk: Risk;
+
+  public isMainPage = true;
+
+  private userID = sessionStorage.getItem(sessionStorageKey.id);
+
+  constructor(
+    private selectedRiskService: SelectedRiskService,
+    private risksSorter: RisksSorterService,
```
Comment (ClainAtomi) [inline risk/src/app/shared/layouts/risks-list/risks-list.component.ts:25]: good job here :)

File: risk/src/app/shared/layouts/site-layout/site-layout.component.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          pButton
+          type="button"
+          label="Main page"
+          (click)="onMainPage()"
+        ></button>
+        <button
+          pButton
+          type="button"
+          label="Logout"
+          (click)="onLogout()"
```
Comment (ClainAtomi) [inline risk/src/app/shared/layouts/site-layout/site-layout.component.html:24]: user router link is better than separate method for each route

File: risk/src/app/shared/layouts/site-layout/site-layout.component.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  public onManagePage(): void {
+    this.router.navigate(['manage']);
+    this.selectedRiskService.selectRisk(defaultRisk);
+    this.isMainPage = !this.isMainPage;
+  }
+
+  public onMainPage(): void {
+    this.router.navigate(['main']);
+    this.selectedRiskService.selectRisk(defaultRisk);
```
Comment (ClainAtomi) [inline risk/src/app/shared/layouts/site-layout/site-layout.component.ts:32]: navigation bar should be encapsulated from other parts of application, it doesn't make sense to modify selected risk from navigation bar itself

File: risk/src/app/shared/services/data-base.service.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    this.httpClient
+      .delete(`${configAPI.urlRisksDatabase}${riskID}`)
+      .subscribe(() => {
+        this.getRiskList();
+      });
+  }
+
+  public updateRiskList(risk: Risk): void {
+    if (risk.id === 'newRisk') {
+      this.httpClient
```
Comment (ClainAtomi) [inline risk/src/app/shared/services/data-base.service.ts:46]: move entity api calls to repository

File: risk/src/app/login-page/login-page.component.html [hunk 7]
```diff
@@ -0,0 +1,83 @@
+<p-card [style]="{ height: '450px' }">
+  <div class="input--block">
+    <h3>Login</h3>
+    <div>
+      <span class="ui-float-label">
+        <input
+          class="ng-dirty ng-invalid"
```
[the remark this answers is not in the document]
Comment (rat6666) [reply risk/src/app/login-page/login-page.component.html:7]: done

File: risk/src/app/login-page/login-page.component.html [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+          text="Password is required"
+        ></p-message>
+        <p-message
+          *ngIf="pass.errors.minlength"
+          severity="error"
+          text="Password must be at least 6 characters long."
+        ></p-message>
+      </div>
+    </div>
+    <p-message
```
[the remark this answers is not in the document]
Comment (rat6666) [reply risk/src/app/login-page/login-page.component.html]: done

File: risk/src/app/main-page/main-page.component.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  selector: 'app-main-page',
+  templateUrl: './main-page.component.html',
+  styleUrls: ['./main-page.component.scss'],
+})
+export class MainPageComponent implements OnInit {
+  constructor(private getSelectedRisk: SelectedRiskService) {}
+
+  risk: Risk = defaultRisk;
+  ngOnInit() {
+    this.getSelectedRisk.selectedRisk$.subscribe((data) => {
```
[the remark this answers is not in the document]
Comment (rat6666) [reply risk/src/app/main-page/main-page.component.ts:17]: done

File: risk/src/app/shared/classes/auth.guard.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  providedIn: 'root',
+})
+export class AuthGuard implements CanActivate, CanActivateChild {
+  constructor(private auth: AuthService, private router: Router) {}
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Observable<boolean> {
+    if (this.auth.isAuthenticate) {
```
[the remark this answers is not in the document]
Comment (rat6666) [reply risk/src/app/shared/classes/auth.guard.ts:14]: done

File: risk/src/app/login-page/login-page.component.scss [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  width: 312px;
+  flex-direction: row;
+  justify-content: space-between;
+}
+
+input {
+  width: 100%;
+}
+
+p-message {
```
[the remark this answers is not in the document]
Comment (rat6666) [reply risk/src/app/login-page/login-page.component.scss]: done

File: risk/src/app/shared/layouts/site-layout/site-layout.component.ts (hunk 5 above)
Comment (rat6666) [reply risk/src/app/shared/layouts/site-layout/site-layout.component.ts:27]: done

File: risk/src/app/shared/layouts/site-layout/site-layout.component.html (hunk 4 above)
Comment (rat6666) [reply risk/src/app/shared/layouts/site-layout/site-layout.component.html]: Here we have to call the logOut() method before the route

File: risk/src/app/shared/interfaces.ts (hunk 1 above)
Comment (rat6666) [reply risk/src/app/shared/interfaces.ts:16]: done

File: risk/src/app/login-page/login-page.component.ts [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+    this.user.password = this.password;
+    this.auth.login(this.user).subscribe(
+      () => this.router.navigate(['main']),
+      () => (this.wrongPassword = true)
+    );
+    this.wrongActionMessageTimer.subscribe(() => (this.wrongPassword = false));
+  }
+
+  public signIn(): void {
+    this.user.username = this.userName;
```
[the remark this answers is not in the document]
Comment (rat6666) [reply risk/src/app/login-page/login-page.component.ts:35]: done

File: risk/src/app/shared/layouts/risk-edit/risk-edit.component.ts (hunk 2 above)
Comment (rat6666) [reply risk/src/app/shared/layouts/risk-edit/risk-edit.component.ts:59]: done

File: risk/src/app/shared/services/data-base.service.ts (hunk 6 above)
Comment (rat6666) [reply risk/src/app/shared/services/data-base.service.ts]: done

**Pull request merged**